src/mmrelay/matrix/e2ee_identity.py

```
@@ -77,21 +90,286 @@ async def _server_has_own_cross_signing_identity(client: object) -> bool:
+            reason,
+        )
+        return False
+
+    logger.info(
+        "Repaired server-visible Matrix self-signing for device %s",
+        _client_label(client, "device_id"),
+    )
+    return True
+
+
 async def _ensure_own_device_cross_signed(
```

Ruff (0.16.1)

[warning] Too many return statements (13 > 6)

(PLR0911)